plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java

```diff
@@ -314,7 +318,49 @@ private String getTargetCompatibility(Project project, SourceSet sourceSet) {
   private List<String> getCompilerArgs(Project project, SourceSet sourceSet) {
     JavaCompile javaCompile = getJavaCompileTask(project, sourceSet);
     if (javaCompile != null) {
-      return javaCompile.getOptions().getCompilerArgs();
+      CompileOptions options = javaCompile.getOptions();
+
+      try {
+        DefaultJavaCompileSpec specs = new DefaultJavaCompileSpec();
+        specs.setCompileOptions(options);
+
+        if (GradleVersion.current().compareTo(GradleVersion.version("6.6")) >= 0) {
+          if (javaCompile.getOptions().getRelease().isPresent()) {
+            specs.setRelease(javaCompile.getOptions().getRelease().get());
+          }
+        }
+
+        if (specs.getSourceCompatibility() == null && specs.getRelease() == null) {
+          String sourceCompatibility = javaCompile.getSourceCompatibility();
+          if (sourceCompatibility != null) {
+            specs.setSourceCompatibility(sourceCompatibility);
+          }
+        }
+        if (specs.getTargetCompatibility() == null && specs.getRelease() == null) {
+          String targetCompatibility = javaCompile.getTargetCompatibility();
+          if (targetCompatibility != null) {
+            specs.setTargetCompatibility(targetCompatibility);
+          }
+        }
+
+        JavaCompilerArgumentsBuilder builder = new JavaCompilerArgumentsBuilder(specs)
+                .includeMainOptions(true)
+                .includeClasspath(false)
+                .includeSourceFiles(false)
+                .includeLauncherOptions(false);
+        return builder.build();
+      } catch (Exception e) {
+        // DefaultJavaCompileSpec and JavaCompilerArgumentsBuilder are internal so may not exist.
```

Comment:
A thing I want to have some discussion a little bit.

I was intended to remove `getSourceCompatibility()` and `getTargetCompatibility()` with the assumption that the `release/source/target` args are always available with this new approach.

Since it's not for old version Gradle, I cannot simply remove it now. The source and target level are used by the java language server to pick the right project jdk.

Do you think it makes sense to manually add `release/source/target` when internal util not exist?

Reply:
I was trying to avoid extending the spec as much as I can. The `JvmBuildTargetEx` extends the spec and it looks like at least the fields `sourceCompatibility` and `targetCompatibility` are not necessary, because they can be covered by the javac options.

Then, if `JavaCompilerArgumentsBuilder` is available, the `sourceCompatibility` and `targetCompatibility` are available as well. On the other hand, we can manually append `--source/--target` or `--release` to the returned string list. Then it should be ok to remove `sourceCompatibility` and `targetCompatibility` from `JvmBuildTargetEx`. The client can check the javac options and get those information.

It's not necessary to cover that in this PR. We can do that when addressing [#112](https://github.com/microsoft/build-server-for-gradle/issues/112). Meanwhile, please let me know if the above assumption is wrong from you point of view.